src/Models/Link.php

```diff
@@ -290,4 +296,23 @@ private function getLinkTypes(): array
 
         return $types;
     }
+
+    public function getDefaultTitle(): string
+    {
+        // If we have a title or don't have link, we can just bail out without any changes
+        if (!$this->config()->get('use_default_title') || empty($this->getURL()) || $this->Title) {
+            return '';
+        }
+        
+        $defaultLinkTitle = $this->getLinkTypeDefaultTitle();
+
+        $this->extend('updateDefaultLinkTitle', $defaultLinkTitle);
+
+        return $defaultLinkTitle;
+    }
+
+    public function getLinkTypeDefaultTitle(): string
```

Comment:
```suggestion
        $defaultLinkTitle = $this->getDefaultTitle();

        $this->extend('updateDefaultLinkTitle', $defaultLinkTitle);

        return $defaultLinkTitle;
    }

    public function getDefaultTitle(): string
```

Please also update the `getLinkTypeDefaultTitle()` method to be called `getDefaultTitle()` in all the subclasses.

Reply:
Updated